Matlab/Matlab.sublime-syntax

```diff
@@ -48,477 +76,792 @@ contexts:
-          scope: meta.brackets.curly.matlab
-          set: transpose_post_parens
-        - include: allofem
-        - include: end_in_parens
-  end_in_parens:
-    - match: \bend\b
-      comment: end as operator symbol
-      scope: keyword.operator.symbols.matlab
-  escaped_quote:
-    - match: "''"
-      scope: constant.character.escape.matlab
+        - meta_scope: meta.braces.matlab
+        - match: \}
+          scope: punctuation.section.braces.end.matlab
+          set: transpose
+        - include: separator-semicolon
+        - include: last-index
+        - include: expressions
 
-  # Function
-  function:
-    - match: '^\s*(function)\b'
-      captures:
-        1 : keyword.other.matlab
+###[ DECLARATION BLOCKS ]#####################################################
+
+  function-declaration:
+    - match: \bfunction\b
+      scope: keyword.declaration.function.matlab
       push:
-        - match: \b(\w+)\s+(=)
+        - meta_scope: meta.function.matlab
+        - include: line-continuation
+        - match: '{{eol}}'
+          set: function-body-arguments
+        - match: ({{identifier}})\s*(=)
           captures:
-            1: variable.parameter.output.function.matlab
+            1: variable.parameter.output.matlab
             2: keyword.operator.assignment.matlab
-        - match: '\['
-          scope: punctuation.section.brackets.matlab
+        - match: \[
+          scope: punctuation.section.brackets.begin.matlab
           push:
-            - match: \b\w+\b
-              scope: variable.parameter.output.function.matlab
-            - match: ','
-              scope: punctuation.separator.matlab
-            - match: '\]'
-              scope: punctuation.section.brackets.matlab
+            - match: '{{identifier}}'
+              scope: variable.parameter.output.matlab
+            - include: line-continuation
+            - match: '{{eol}}'
+              set: function-body-arguments
+            - include: separator-comma
+            - match: \]
+              scope: punctuation.section.brackets.end.matlab
               pop: true
-        - match: '(=)?\s*\b(\w+)\s*(\()'
+        - match: =
+          scope: keyword.operator.assignment.matlab
+        - match: (?:(?:set|get)(\.))?{{identifier}}
+          scope: entity.name.function.matlab
           captures:
-            1: keyword.operator.assignment.matlab
-            2: entity.name.function.matlab
-            3: punctuation.section.parens.begin.matlab
+            1: punctuation.accessor.dot.matlab
+        - match: (?=\()
           set:
             - meta_scope: meta.function.parameters.matlab
-            - match: \b\w+\b
-              scope: variable.parameter.input.function.matlab
-            - match: ','
-              scope: punctuation.separator.matlab
-            - match: '\)'
+            - include: line-continuation
+            - match: '{{eol}}'
+              set: function-body-arguments
+            - match: \(
```

Comment:
I'm not exactly sure why I used that lookahead here, maybe because it was necessary for correct boundaries of the meta scopes. I'll check it soon and look whether I can adjust it.

Reply:
I've cleaned up the rules for function declarations a bit and eliminated the lookahead. There should be no nested parens for function parameters, but correct balancing will be ensured now, like for the other parens/brackets expressions.